contributors/design-proposals/node/pod-pid-namespace.md

```diff
@@ -1,77 +1,150 @@
-configure the behavior per-pod. We believe this is acceptable because:
+### Configurable PID Namespace Sharing
 
-* We have not identified a concrete use case requiring isolated PID namespaces.
-* Making PID namespace configurable requires changing the CRI, which we would
-  like to avoid since there are no use cases.
+We will add first class support for configuring isolated and shared PID
+namespaces for pods by adding a new boolean field `sharedPID` to the pod spec.
+When set to true, all containers in the pod will share a single PID namespace
+when supported by runtime. When not supported by the runtime, for example with
+docker versions less than 1.13.1, the runtime will respond with an error and the
+pod will not start.
 
-In a future release, SIG Node will recommend docker >= 1.12. Unless a compelling
-use case for isolated PID namespaces is discovered, we will remove the ability
-to disable the shared PID namespace in the subsequent release.
+The Container Runtime Interface (CRI) will be updated to explicitly support
+shared PID namespaces. When `sharedPID` is true, runtimes MUST share a PID
+namespace for the pod or return an error. When `sharedPID` is false, runtimes
+MAY implement isolated PID namespaces. That is, omitting this option defaults to
+pre-existing runtime behavior. For Docker the default remains isolated PID
+namespaces.
 
+The shared PID functionality will be hidden behind a new feature gate in both
+the API server and the kubelet, and the `--docker-disable-shared-pid` flag will
+be removed from the kubelet.
 
-[1]: https://blog.phusion.nl/2015/01/20/docker-and-the-pid-1-zombie-reaping-problem/
+## User Experience
 
+### Use Cases
 
-<!-- BEGIN MUNGE: GENERATED_ANALYTICS -->
-[![Analytics](https://kubernetes-site.appspot.com/UA-36037335-10/GitHub/docs/proposals/pod-pid-namespace.md?pixel)]()
-<!-- END MUNGE: GENERATED_ANALYTICS -->
+Sharing a PID namespace between containers in a pod is discussed in
+[#1615](https://issues.k8s.io/1615) and enables:
+
+1.  signaling between containers, which is useful for side cars (e.g. for
+    signaling a daemon process after rotating logs).
+1.  easier troubleshooting of pods.
+1.  addressing [Docker's zombie
+    problem](https://blog.phusion.nl/2015/01/20/docker-and-the-pid-1-zombie-reaping-problem/)
+    by reaping orphaned zombies in the infra container.
+
+## Implementation
+
+### Kubernetes API Changes
+
+`v1.PodSecurityContext` gains a new field resembling the existing `hostPID`
+field:
+
+```
+type PodSecurityContext struct {
+    ...
+    // Use the host's pid namespace.
+    // Optional: Default to false.
+    // +k8s:conversion-gen=false
+    // +optional
+    HostPID bool
+    // Use a single shared PID namespace for the pod.
+    // Optional: Default to false.
+    // +k8s:conversion-gen=false
+    // +optional
+    SharedPID bool
+       ...
+```
+
+`PodSecurityContext` was chosen because it's where namespace configuration
```

Comment:
I believe this is the internal API object..... The v1.PodSecurityContext does not contain the PID namespace fields...Could you update the text/code here to reflect that?

Reply:
Whoops, you're right. Updated.